src/borg/archiver.py

```diff
@@ -433,6 +435,14 @@ def item_is_hardlink_master(item):
             return (partial_extract and stat.S_ISREG(item.mode) and
                     item.get('hardlink_master', True) and 'source' not in item)
 
+        if progress:
+            extracted_size = 0
+            for item in archive.iter_items(filter=lambda item: matcher.match(item.path)):
+                extracted_size += item.file_size()
```

Comment:
imagine some huge sparse files being in there, like VM disk images.

the progress estimation will be very wrong as the size will look huge, but in the end it will be just a lot of copies of the same all-zero-chunk and (depending on other flags) might just result in a lot of seeks on the output file.

even without sparse files: if a file has a lot of duplicated chunks internally, we could optimize that and make it much faster than without duplication (not sure if we already do that).

so, I am still rather against progress indication that outputs information that can be very wrong and would prefer outputting only information that is correct for sure.